client/src/modules/stock/entry/entry.js

```diff
@@ -676,24 +688,32 @@ function StockEntryController(
    * @description [grid] initialize each cell of defined rows with value
    */
   function buildStockLine(line) {
-    const inventory = inventoryStore.get(line.inventory_uuid);
-    const entryDate = vm.movement.date || Date();
-    line.code = inventory.code;
-    line.label = inventory.label;
-    line.unit_cost = inventory.price;
-    line.quantity = 0;
-    line.cost = line.quantity * line.unit_cost;
-    line.expiration_date = entryDate;
-    line.unit = inventory.unit;
-    line.tracking_expiration = inventory.tracking_expiration;
-    setInitialized(line);
 
     // Store the non-expired candidate lots for this inventory code
     Lots.candidates({ inventory_uuid : line.inventory_uuid, date : vm.movement.date })
       .then((lots) => {
         line.availableLots = lots;
         line.candidateLots = lots.filter(lot => !lot.expired);
-      });
+
+        const inventory = inventoryStore.get(line.inventory_uuid);
+        const entryDate = vm.movement.date || Date();
+        line.code = inventory.code;
+        line.label = inventory.label;
+        line.unit_cost = inventory.price;
+        line.quantity = 0;
+        line.cost = line.quantity * line.unit_cost;
+        line.expiration_date = entryDate;
+        line.unit = inventory.unit;
+        line.tracking_expiration = inventory.tracking_expiration;
+        setInitialized(line);
+
+        return Inventory.inventoryWac(line.inventory_uuid);
```

Comment:
This looks like something that we should on the server before sending the lots or inventory items back to the client.